src/Rules/Nofib.hs

```diff
@@ -0,0 +1,55 @@
+module Rules.Nofib where
+
+import Base
+import Expression
+import GHC
+import Oracles.Setting
+import Target
+import Utilities
+
+import System.Environment
+
+nofibRules :: Rules ()
+nofibRules = do
+  root <- buildRootRules
+
+  -- a phony "nofib" rule that just triggers
+  -- the rule below.
+  "nofib" ~> need [root -/- nofibLogFile]
+
+  -- a rule to produce <build root>/nofig-log
+  -- by running the nofib suite and capturing
+  -- the relevant output.
+  root -/- nofibLogFile %> \fp -> do
+    needNofibDeps
```

Comment:
I'm confused: why don't you simply execute all required commands here? What is the benefit of having a separate builder?

In the rest of Hadrian, we use builders primarily to have a unified way to track builder-specific command line arguments. But you are not using `Args` here, so it seems that defining a new builder is not required.

Reply:
OK, I'll simplify this.

Reply:
Done.